Doc/tools/extensions/changes.py

```diff
@@ -54,16 +71,23 @@ def run(self) -> list[Node]:
             raise ValueError(
                 "deprecated-removed:: second argument cannot be `next`"
             )
-        self.arguments[0] = version_deprecated, version_removed
+        self.arguments[0] = TerribleHashableDictHack(
```

Comment:
Another option that avoids this hack is to check the order of `{deprecated}` and `{removed}` in the string, adjust the order of the tuple accordingly, and replace each placeholder with `%s`.  That precludes the possibility of a translation needing to repeat either placeholder, but I don't know if that's a realistic need.

Note that the not-a-`dict` hack is necessary for this route due to `dict` not being hashable, raising a `SphinxParallelError`.